src/ralph/backends/data/base.py

```diff
@@ -383,22 +387,40 @@ async def write(  # noqa: PLR0913
             operation_type = self.default_operation_type
 
         if operation_type in self.unsupported_operation_types:
-            msg = f"{operation_type.value.capitalize()} operation_type is not allowed."
+            msg = f"{operation_type.value.capitalize()} operation_type is not allowed"
             self.logger.error(msg)
             raise BackendParameterException(msg)
 
         data = iter(data)
         try:
             first_record = next(data)
         except StopIteration:
-            self.logger.info("Data Iterator is empty; skipping write to target.")
+            self.logger.info("Data Iterator is empty; skipping write to target")
             return 0
         data = chain((first_record,), data)
 
         chunk_size = chunk_size if chunk_size else self.settings.WRITE_CHUNK_SIZE
         is_bytes = isinstance(first_record, bytes)
         writer = self._write_bytes if is_bytes else self._write_dicts
-        return await writer(data, target, chunk_size, ignore_errors, operation_type)
+
+        concurrency = concurrency if concurrency else 1
+        if concurrency == 1:
+            return await writer(data, target, chunk_size, ignore_errors, operation_type)
+
+        if concurrency < 1:
```

Comment:
We can type `concurrency` with `PositiveInt` from `pydantic` in the method signature. WDYT?

Reply:
Good Idea) Thanks) Updated it)